### DISCUSSION TITLE
Missing string data from Inverter 2

### DISCUSSION THREAD

**Discussion by amurug** [jasonacox/Powerwall-Dashboard#502] [Q&A]
Hi,

I've been using Powerwall-Dashboard for about a year now (since I got my Tesla solar panels / Powerwalls). It's been an amazing software to use; appreciate you guys !

While for the first 6-9 months, I had visibility to all of my string data (have 4 strings on each inverter; and would show up as A/B/C/D and A1/B1/C1/D1) and both inverters (Powerwall 2).. I have recently run into an issue where the string data completely disappeared and only the power wall charge was being updated.

Tried to uninstall and reinstall the power wall-dashboard a few times, and now I was able to get some string data, but it is only tied to the first inverter (A/B/C/D). After multiple tries, I am still unable to get the 8 strings and 2 inverters to show as it used to before (I have 4 strings and 1 inverter data show up on my dashboard).

Please advise if I need to do something different to get visibility to the other inverter and set of strings ?

Thank you !

**Comment by jasonacox** [comment]
Hi @amurug 

Several months ago, Tesla introduced a FIrmware upgrade that removed the `vitals` API which is how we used to get the string data, alerts and other nice metrics. We found an API that provides the first inverter data as you noticed. (Related Issue https://github.com/jasonacox/Powerwall-Dashboard/issues/436).

Most recently, we discovered a new API (called TEDAPI - see https://github.com/jasonacox/Powerwall-Dashboard/discussions/392) that is only available on the Gateway WiFi access point (192.168.91.1) that provides most of the vitals data we had before. The Dashboard now supports this and will provide the extended data you are looking for. However, it requires some additional steps (see [notes](https://github.com/jasonacox/Powerwall-Dashboard?tab=readme-ov-file#extended-device-vitals-metrics-beta)), specifically:

1. Set up routing from your Dashboard host to the Powerwall WiFi gateway:


```
# Example - Change 192.168.0.100 to the IP address of Powerwall Gateway on your LAN

# Linux Ubuntu and RPi - Can add to /etc/rc.local for persistence 
sudo ip route add 192.168.91.1 via 192.168.0.100

# MacOS 
sudo route add -host 192.168.91.1 192.168.0.100 # Temporary 
networksetup -setadditionalroutes Wi-Fi 192.168.91.1 255.255.255.255 192.168.0.100 # Persistent

# Windows - Using persistence flag - Administrator Shell
route -p add 192.168.91.1 mask 255.255.255.255 192.168.0.100
```

2. Configure `pypowerwall.env` with the Gateway WiFi Password (usually on the QR code on the Powerwall Gateway) or re-run `setup.sh` to re-configure your setup. The manual step would be to edit `pypowerwall.env` and add:

```
PW_GW_PWD=PASSWORD
```

Where `PASSWORD` is the Gateway WiFi password (on the QR code).

* Restart your Dashboard with `./compose-dash.sh up -d`.
* If you have the latest Dashboard code, you can run verify to see if it is connecting: `verify.sh`

**Reply by amurug** [reply]
Thank you so much Jason. Appreciate your response & guidance.

I was trying to setup routing on my Unifi router as per your suggestion, but not sure if I am doing this right. 

I have a home network with separate VLANs:
192.168.1.x - Primary
192.168.3.x - IOT

I have the Tesla Gateway connected to the IOT VLAN both wired and wireless with a 192.168.3.x IP address.

I understand the Tesla Gateway itself has a DHCP server (when connecting via the TEG-xxx wifi network) on the 192.168.91.x network.

Is there still a way to bridge / setup routing between the machine which has the powerwall-dashboard running (on the 192.168.1.x network) and the 192.168.91.1 ? Or do I need the machine to be connected to the TEG wifi network for this to work ?

Appreciate your help again. Thanks

**Reply by jgleigh** [reply]
All you need is a static route. I have mine setup as Destination: 192.168.91.0/24 and Next Hop as the local IP address (wired probably preferred).

Depending on how isolated your VLANs are, you might need to modify your firewall rules too.

**Reply by jasonacox** [reply]
@jgleigh is right, I think you might be making it more complex than it needs to be. If the Powerwall has a local LAN address (your 192.168.1.x ?) you would just set up a route on your host that is on that same LAN to route 192.168.91.1 traffic via the Powerwall's local LAN address:

```bash
sudo ip route add 192.168.91.1 via 192.168.1.x # (where x is the address of the Powerwall)

# test from same host
ping -c 1 192.168.91.1
```

```
 ___________________           __________________________       
[ Powerwall Gateway ]         [      Dashboard Host      ]    
[       TEG         ]   LAN   [__________________________] 
[ LAN: 192.168.1.x  ] <-----  [     LAN: 192.168.1.12    ] 
[___________________] or WLAN [__________________________]
[   192.168.91.1    ]         [     ip route add...      ]     
 ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾           ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾       
```

**Comment by amurug** [comment]
Thank you Jason and Jeff. I will give that a try. Appreciate your guidance. On Jul 18, 2024, at 8:42 PM, Jason Cox ***@***.***> wrote:﻿
@jgleigh is right, I think you might be making it more complex than it needs to be. If the Powerwall has a local LAN address (your 192.168.1.x ?) you would just set up a route on your host that is on that same LAN to route 192.168.91.1 traffic via the Powerwall's local LAN address:
sudo ip route add 192.168.91.1 via 192.168.1.x # (where x is the address of the Powerwall)

# test from same host
ping -c 1 192.168.91.1

**Comment by amurug** [comment]
Thanks again Jason and Jeff. Was able to test with temporary route and then modify to a persistent route. The extended stats are working great. 

Appreciate your help !

**Reply by jgleigh** [reply]
I'm running a UDM Pro so feel free to ask if you have any problems with the routing or firewall rules.
